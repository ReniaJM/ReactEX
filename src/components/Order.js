import React from 'react'
import OrderView from './OrderView'


class Order extends React.Component{
    render(){
        const orderedbooks = this.props.order.map(order =>{
            return <OrderView book = {order} removeOrder={ this.props.removeOrder}/>
        })
        let adminCss = {
            paddingTop: '15px',
        }
        return (
            <div className = "order col-md-6" style={adminCss}>
                <h2>Your order:</h2>
                {orderedbooks}
            </div>
        )
    }
}

export default Order;