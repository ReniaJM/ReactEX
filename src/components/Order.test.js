import ReactDOM from "react-dom";
import React from "react";
import Order from "./Order.js";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';


configure ({adapter: new Adapter ()});

describe ('Order-test', ()=> {
    it('Order render without a problem', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Order/>, div);
        ReactDOM.unmountComponentAtNode(div);
        // to jets kasowanie diva
    });

    it('hello world renderee', () => {
        const wrapper = shallow(<Order/>);
        // console.log(wrapper.debug());
        expect(wrapper.find('div').text('ZamÃ³wienie'))
    })
    it('snapshot matches', () => {
        const wrapper = shallow(<Order/>);
        // console.log(wrapper.debug());
        expect(wrapper).toMatchSnapshot();
    })

})

