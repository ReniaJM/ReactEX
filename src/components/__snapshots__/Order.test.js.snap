// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order-test snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Order />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Zam贸wienie",
      "className": "order col-md-4",
      "style": Object {
        "paddingTop": "15px",
      },
    },
    "ref": null,
    "rendered": "Zam贸wienie",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Zam贸wienie",
        "className": "order col-md-4",
        "style": Object {
          "paddingTop": "15px",
        },
      },
      "ref": null,
      "rendered": "Zam贸wienie",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
